WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingBottomSheetFragment.kt

```diff
@@ -119,15 +125,37 @@ class PrepublishingBottomSheetFragment : WPBottomSheetDialogFragment(),
     }
 
     /**
-     * On Android 10, the bottom sheet is not above the window inset so that leads to the publish button being cut
-     * off. To fix this, we add the bottom inset as the margin bottom of the fragment container.
+     * On Android 10 and above, in IMMERSIVE/FULLSCREEN mode, the bottom sheet is not above the window inset so that
+     * leads to the publish button being cut off. To fix this, we add the bottom inset as the margin bottom of the
+     * fragment container. When the Navigation Mode is changed to Gesture Mode, the bottom inset does not incorporate
+     * the height of the status bar  ( or the total area that should represent the bounds of navigation  area.)
+     * so currently it is being added dynamically. This solution will be improved as soon as the Android team
+     * releases an update to the insets API.
      */
     private fun PostPrepublishingBottomSheetBinding.addWindowInsetToFragmentContainer() {
-        if (VERSION.SDK_INT >= VERSION_CODES.Q) {
-            activity?.window?.decorView?.rootWindowInsets?.systemWindowInsetBottom?.let { bottomInset ->
-                val param = prepublishingContentFragment.layoutParams as ViewGroup.MarginLayoutParams
-                param.setMargins(0, 0, 0, bottomInset)
-                prepublishingContentFragment.layoutParams = param
+        // utilized to run the logic with the StoryComposerActivity since it is in IMMERSIVE mode.
+        val isStoryPost = checkNotNull(arguments?.getBoolean(IS_STORY_POST)) {
+            "arguments can't be null."
+        }
+
+        if (VERSION.SDK_INT >= VERSION_CODES.Q && isStoryPost) {
+                activity?.window?.decorView?.rootWindowInsets?.let { windowInsets ->
+                    val param = prepublishingContentFragment.layoutParams as ViewGroup.MarginLayoutParams
+                    if (navBarModeHelper.getNavigationMode(requireActivity())
+                                    ?.equals(NavigationMode.NAV_BAR_MODE_GESTURAL) == true) {
+                        param.setMargins(
+                                0,
+                                0,
+                                0,
+                                windowInsets.systemGestureInsets.bottom + dpToPx(
```

Comment:
I utilized the `systemGestureInsets` because it was returning an inset that represents the touchable area for gestures. Using `systemWindowInsetBottom` here had resulted in an inset that was really low, which I think is due to the fact that the host/parent activity is in immersive mode.